src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -131,7 +157,19 @@ public void cleanup() {
             for (Map.Entry<View, LayoutRule> original:mOriginalValues.entrySet()) {
                 final View changedView = original.getKey();
                 final LayoutRule originalValue = original.getValue();
-                setLayout(changedView, originalValue.verb, originalValue.anchor);
+                try {
+                    setLayout(changedView, originalValue.verb, originalValue.anchor);
+                } catch (CantVisitException e) {
+                    ; // shouldn't reach here
+                }
+            }
+            mAlive = false;
+        }
+
+        @Override
+        public void visit(View rootView) {
+            if (mAlive) {
+                super.getPathfinder().findTargetsInRoot(rootView, super.getPath(), this);
```

Comment:
I don't think you need the `super` here, the methods will just resolve (and that way, if we ever override them it'll just work :) )